app/controllers/microposts_controller.rb

```diff
@@ -0,0 +1,38 @@
+class MicropostsController < ApplicationController
+  before_action :logged_in_user, only: %i(create destroy)
+  before_action :correct_user, only: :destroy
+
+  def create
+    @micropost = current_user.microposts.build micropost_params
+
+    if @micropost.save
+      flash[:success] = t(".micro.created")
+      redirect_to root_url
+    else
+      @feed_items = []
+      flash[:danger] = t(".micro.nocreated")
+      render "static_pages/home"
+    end
+  end
+
+  def destroy
+    if @micropost.destroy
+      flash[:success] = t("microposts.controller.delete")
+      redirect_to request.referrer || root_url
+    else
+      flash[:danger] = t("microposts.controller.undelete")
+      redirect_to request.referer
```

Comment:
request.referer trả về cái gì em

Reply:
referer trả về link trước đó. Ví dụ khi anh ấn xoá micropost thì xoá thành công nó vẫn ở lại trang này. This uses the request.referrer method,12 which is related to the request.original_url variable used in friendly forwarding (Section 10.2.3), and is just the previous URL (in this case, the Home page).13 This is convenient because microposts appear on both the Home page and on the user’s profile page, so by using request.referrer we arrange to redirect back to the page issuing the delete request in both cases.